src/modules/Rooms/Components/RoomsForm/RoomsForm.tsx

```diff
@@ -1,6 +1,443 @@
+  facilities: string[] | RoomFacilities[];
+  imgs: File[];
+  images?: string[];
+};
 export default function RoomsForm() {
+  const inputRef = useRef<HTMLInputElement>(null);
+  const {
+    register,
+    formState: { errors, isSubmitting },
+    handleSubmit,
+    getValues,
+    setValue,
+    trigger,
+    watch,
+    control,
+  } = useForm<Room>({
+    defaultValues: {
+      imgs: undefined,
+      roomNumber: "",
+      price: "",
+      capacity: "",
+      discount: "",
+      facilities: [],
+    },
+    mode: "onChange",
+  });
+
+  const navigate = useNavigate();
+  const facilities = watch("facilities");
+  const imgs = getValues("imgs");
+  const uploadedImgs = getValues("images");
+  const validationRules = getRoomValidationRules();
+  const url = useObjectUrl(imgs);
+  const { roomId } = useParams();
+  const { pathname } = useLocation();
+  const newRoom = pathname.includes("new-room");
+
+  const onSubmit = async (data: Room) => {
+    console.log(data);
+    const formData = new FormData();
+    for (const key in data) {
+      if (key === "imgs") {
+        if (uploadedImgs) {
+          const filesFromUrls = await convertUrlsToFiles(uploadedImgs || []);
+          filesFromUrls.forEach((file) => formData.append("imgs", file));
+        }
+        const fileList = data.imgs;
+        const files = Array.from(fileList);
+        for (const file of files) {
+          formData.append("imgs", file);
+        }
+      } else if (key === "facilities") {
+        if (
+          Array.isArray(data[key]) &&
+          data[key].every((facility) => typeof facility === "string")
+        ) {
+          // facilities is an array of strings
+          data[key].forEach((facility: string) => {
+            formData.append("facilities", facility);
+          });
+        }
+      } else {
+        formData.append(key, data[key as keyof Room] as string);
+      }
+    }
+
+    try {
+      const response = await axiosInstance?.[
+        newRoom ? "post" : "put"
+      ]<RoomResponse>(
+        newRoom
+          ? ROOM_URLS.createRoom
+          : `${ROOM_URLS.updateRoom(roomId ?? "")}`,
+        formData
+      );
+      if (response.status === 201) {
+        toast.success(response?.data?.message || "Room created successfully");
+        navigate("/rooms");
+      } else if (response.status === 200) {
+        toast.success(response?.data?.message || "Room updated successfully");
+        navigate("/rooms");
+      }
+    } catch (error: unknown) {
+      if (axios.isAxiosError(error)) {
+        toast.error(
+          error.response?.data?.message ||
+            "Failed to create room. Please try again."
+        );
+      } else {
+        console.error(error);
+      }
+    }
+  };
+  const handleRemoveImage = (index: number) => {
+    const updatedFiles = imgs && imgs!.filter((_, i) => i !== index);
+    setValue("imgs", updatedFiles, { shouldValidate: true });
+  };
+  async function convertUrlsToFiles(urls: string[]) {
```

Comment:
I thought that we didn't handle urls conversion to blobs, but it seems that they are not working when deployed because the backend return an http url in the room response which is unsecure and blocked. I checked changing the url to https instead and it was working. We can replace the http in any url coming from backend to https and should work as well.
![image](https://github.com/user-attachments/assets/e6b7d5a2-91cc-48e8-8da7-e38ea3b972c6)